packages/amplify-graphql-auth-transformer/src/utils/index.ts

```diff
@@ -0,0 +1,101 @@
+import { ModelDirectiveConfiguration, SubscriptionLevel } from '@aws-amplify/graphql-model-transformer';
+import { AppSyncAuthConfiguration, AppSyncAuthMode, DirectiveWrapper } from '@aws-amplify/graphql-transformer-core';
+import { ObjectTypeDefinitionNode, FieldDefinitionNode, DirectiveNode } from 'graphql';
+import { toCamelCase } from 'graphql-transformer-common';
+import { AuthProvider, AuthRule } from './definitions';
+
+export * from './constants';
+export * from './definitions';
+export * from './validations';
+
+/**
+ * Ensure the following defaults
+ * - provider
+ * - iam policy generation
+ */
+export const ensureAuthRuleDefaults = (rules: AuthRule[]) => {
+  // We assign the default provider if an override is not present make further handling easier.
+  for (const rule of rules) {
+    if (!rule.provider) {
+      switch (rule.allow) {
+        case 'owner':
+        case 'groups':
+          rule.provider = 'userPools';
+          break;
+        case 'private':
+          rule.provider = 'userPools';
+          break;
+        case 'public':
+          rule.provider = 'apiKey';
+          break;
+        default:
+          rule.provider = null;
+          break;
+      }
+    }
+    if (rule.provider === 'iam') {
+      rule.generateIAMPolicy = true;
+    }
+  }
+};
+
+export const getModelConfig = (directive: DirectiveNode, typeName: string): ModelDirectiveConfiguration => {
+  const directiveWrapped: DirectiveWrapper = new DirectiveWrapper(directive);
+  const options = directiveWrapped.getArguments<ModelDirectiveConfiguration>({
+    queries: {
+      get: toCamelCase(['get', typeName]),
+      list: toCamelCase(['list', `${typeName}s`]),
```

Comment:
Can you use the new pluralization logic. https://github.com/aws-amplify/amplify-cli/commit/38e25996ee00479031c88714af3b9d40ef9e079c has examples.